model/registry/db.yaml

```diff
@@ -529,3 +533,67 @@ groups:
           reference the [elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json)
           in order to map the path part values to their names.
         examples: ['db.elasticsearch.path_parts.index=test-index', 'db.elasticsearch.path_parts.doc_id=123']
+  - id: registry.db.vector
+    prefix: db.vector
+    type: attribute_group
+    brief: >
+      This group defines attributes for vector databases.
+    attributes:
+      - id: similarity
+        type:
+          allow_custom_values: true
+          members:
+            - id: cosine
+              value: 'cosine'
+              brief: >
+                The cosine metric.
+              stability: experimental
+            - id: dot
+              value: 'dot'
+              brief: >
+                The dot product metric.
+              stability: experimental
+            - id: euclidean
+              value: 'euclidean'
+              brief: >
+                The euclidean distance metric.
+              stability: experimental    
+            - id: manhattan
+              value: 'manhattan'
+              brief: >
+                The Manhattan distance metric.
+              stability: experimental
+        stability: experimental
+        brief: >
+          The metric used in similarity search.
+        examples: 'cosine'
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          The ID of vector.
+        examples: '5c56c793-69f3-4fbf-87e6-c4bf54c28c26'
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          The name field as of the vector (e.g. a field name).
+        examples: 'vector'
+      - id: embeddings
+        type: double[]
+        stability: experimental
+        brief: >
+          The values of the vector, the array of numbers.
+        examples: [0.9, 0.1, 0.1]
+      - id: dimension
+        type: int
+        stability: experimental
+        brief: >
+          The dimension of the vector.
+        examples: [3]
+      - id: model
```

Comment:
this should be captured with `gen_ai.request.model` (and other gen-ai attributes) - it's ok to mix different attributes on the same telemetry item.

Reply:
This is the `model` for the embedding and can be different from the `gen_ai.request.model`. Vector databases are not strictly related with GenAI. There are many use cases where vector db can be used without GenAI (e.g. semantic search).

Reply:
could you provide some examples of databases that can *compute* embeddings? 

How would database instrumentation know which model was used to create embeddings if it only stores and queries them?

Reply:
Some vendors that provide in-database embedding:

- [Weaviate](https://weaviate.io/), supporting the following [provider integrations](https://weaviate.io/developers/weaviate/model-providers);
- [Elasticsearch](https://www.elastic.co/), using the [Inference API](https://www.elastic.co/guide/en/elasticsearch/reference/current/inference-apis.html);
- [PostgresML](https://postgresml.org/) provides [in-database embedding generation](https://postgresml.org/docs/open-source/pgml/guides/embeddings/in-database-generation);
- [Vespa](https://vespa.ai/) provides the embedding using the [embed()](https://docs.vespa.ai/en/embedding.html) function;
- [Supabase](https://supabase.com/) provides an embegging generator using [edge function](https://supabase.com/docs/guides/ai/quickstarts/generate-text-embeddings);

I think many other vendors will add this feature. Having the possibility to generate embeddings in the database simplify the customer use cases.

Regarding the question on how database instrumentation know which model was used to create an embedding this is an information that is specified when you create a collection or when you provide some search (e.g. in [PostgreML](https://postgresml.org/docs/open-source/pgml/guides/embeddings/in-database-generation#creating-embeddings)). Moreover, I think the instrumentation libraries can also leverage this `model` attribute since they know the embedding model used to generate the vectors.

Reply:
It seems all of them use some GenAI model (in-process or external) and provide an integration layer with it. 

I'm not sure what benefit defining a new attribute for databases brings. If embeddings become cross-domain concerns, let's find a generic attribute name for the vectorization model that will be reused between DB and GenAI. For now I strongly recommend to avoid adding new attribute.